backend/app/server/teams.go

```diff
@@ -58,89 +72,91 @@ func (server *Server) signupsAllowed(eventId string) bool {
 	}
 }
 
+// for path teams/browse
 func (server *Server) GetAllTeams(ctx *gin.Context) {
+	// what if there's no session => no user id
 	teams, err := database.GetTeams()
 	if err == nil {
 		ctx.JSON(http.StatusOK, teams)
 	} else {
+		fmt.Println("ERROR: ", err)
 		ctx.Status(http.StatusInternalServerError)
+		return
 	}
 }
 
 func (server *Server) GetUserTeams(ctx *gin.Context) {
 	session := sessions.Default(ctx)
 	userId := session.Get("userId")
-	strUserId := userId.(string)
 
 	if userId == nil {
 		ctx.Status(http.StatusNotFound)
 		return
 	}
-	// var teamResponse GetTeamResponse
-	// var teamsResponse []GetTeamResponse
+	strUserId := userId.(string)
 
 	teams, err := database.GetUserTeams(convert.StringToUUID(strUserId))
 	if err != nil {
 		ctx.Status(http.StatusInternalServerError)
 		return
 	}
 
-	//1. join databse to return members
 	ctx.JSON(http.StatusOK, teams)
-
-	// add all the GetTeamResponse to []GetTeamResponse
-	// loop through teams, get team id
-	// assign each team to teamResponse type..
-
 }
 
-// stepp 4: GET team info
 // purpose is to construct the DBTeamMemberInfo
-func (server *Server) GetTeamInfo(ctx *gin.Context) {
-	id := convert.StringToUUID(ctx.Param("id"))
+func (server *Server) GetTeamInfo(id pgtype.UUID) (*GetTeamResponse, error) {
 
 	var teamResponse GetTeamResponse
 	var team database.DBTeam
 	var event database.DBEvent
-	var members *[]database.DBTeamMemberInfo //user info based on teamId
-
+	var teamMembers *[]database.DBTeamMemberInfo //user info based on teamId
+	// fmt.Println("========id: ", id) prints: {[204 69 126 62 33 10 77 93 131 216 8 153 66 109 252 147] true}
 	team, err := database.GetTeam(id)
 	if err != nil {
 		logger.Error("failed to get team: %v", err)
-		ctx.JSON(http.StatusInternalServerError, gin.H{"error": fmt.Sprintf("failed to get team: %v", err)})
-		return
+		return nil, err
 	}
 
 	event, err = database.GetEvent(team.EventId)
 	if err != nil {
 		logger.Error("failed to get event: %v", err)
-		ctx.JSON(http.StatusInternalServerError, gin.H{"error": fmt.Sprintf("failed to get event: %v", err)})
-		return
+		return nil, err
 	}
 
-	members, err = database.GetMembersByTeamId(team.Id)
+	teamMembers, err = database.GetMembersByTeamId(team.Id)
 	if err != nil {
 		logger.Error("failed to get event: %v", err)
-		ctx.JSON(http.StatusInternalServerError, gin.H{"error": fmt.Sprintf("failed to get members: %v", err)})
-		return
+		return nil, err
 	}
 
 	// attach all 3 structures to GetTeamResponse --> nested structs turn into nested JSON (with ctx.JSON)
 	teamResponse.Team = &team
 	teamResponse.Event = &event
-	teamResponse.Members = members
+	teamResponse.TeamMembers = teamMembers
+
+	return &teamResponse, nil
+}
+
+func (server *Server) sendTeamInfo(ctx *gin.Context) {
```

Comment:
This should be called GetTeamInfo to follow the standard of everything else, as its the handler a GET request.  

I assume it's named this way because there's another GetTeamInfo method already.  I would rename that existing method to something like AggregateTeamInfo or AssembleTeamInfo since it's responsible for pulling together a bunch of different pieces into a larger structure representing the team.